girder/utility/filesystem_assetstore_adapter.py

```diff
@@ -303,7 +316,12 @@ def deleteFile(self, file):
         if matching.count(True) == 1:
             path = os.path.join(self.assetstore['root'], file['path'])
             if os.path.isfile(path):
-                os.remove(path)
+                with filelock.FileLock(path + '.deleteLock'):
```

Comment:
When we have an upload, we obtain the lock just long enough to check if the file exists.  If so, then the file exists and was not being actively deleted, so we are safe to delete the temp file -- we've already marked that the upload is using the file.

When we are remove a Girder file, if it is the only girder file using the physical file because no girder file lists it, the we get a lock, make sure no upload is using it, and then remove it.  I don't see an order when we would accidentally delete an upload.  However, if an upload has completed, our file could could change to 2 in the interim between checking how many files are using the physical file and the delete.

So, yes, we need to move the lock, but I think for different reasons.